contracts/GroupVoteLocker.sol

```diff
@@ -0,0 +1,331 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import "./BaseVoteLocker.sol";
+import "OpenZeppelin/openzeppelin-contracts@02fcc75bb7f35376c22def91b0fb9bc7a50b9458/contracts/utils/math/SafeCast.sol";
+import "OpenZeppelin/openzeppelin-contracts@02fcc75bb7f35376c22def91b0fb9bc7a50b9458/contracts/utils/Strings.sol";
+import "./console.sol";
+
+contract GroupVoteLocker is BaseVoteLocker {
+    ///@notice Checkpoint structure representing linear voting power decay
+    struct Checkpoint {
+        int128 bias;
+        int128 slope;
+        uint256 ts;
```

Comment:
64 bit timestamps are already good until the sun explodes. If we switch this to 128 bit, and the block number to 128 bit, then we will save a storage slot here in the struct that is going to have tons of reads and writes.